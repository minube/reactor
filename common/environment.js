import { DeviceInfo, Platform } from 'react-native';
import PKG from '../package.json';

const { NODE_ENV, REACT_APP_ENV = 'development' } = process.env;
const IS_SERVER = typeof window === 'undefined';
const IS_WEB = Platform.OS === 'web';
const IS_PRODUCTION = NODE_ENV === 'production' && NODE_ENV === REACT_APP_ENV;
const IS_MOBILE_WEB = IS_WEB && !IS_SERVER
  ? /iPhone|Android|Tablet|iPad|Mobile|iPod|Windows Phone|Blackberry/i.test(DeviceInfo.userAgent)
  : false;

if (IS_WEB && !IS_SERVER) window.reactor = { version: PKG.version };

export default {
  IS_NATIVE: !IS_WEB,
  IS_SERVER: IS_SERVER && IS_WEB,
  IS_WEB,
  IS_MOBILE_WEB,
  IS_PRODUCTION,
  IS_DEVELOPMENT: !IS_PRODUCTION,
  IS_TEST: NODE_ENV === 'test',

  NODE_ENV,

  PKG: {
    name: PKG.name,
    version: PKG.version,
  },

  REACT_APP_ENV,
};
