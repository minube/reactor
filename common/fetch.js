const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
  'X-Requested-With': 'XMLHttpRequest',
};

export default async (endpoint, props = {}) => {
  const { headers, method = 'GET', ...others } = props;

  return new Promise((resolve, reject) => {
    fetch(endpoint, {
      headers: { ...DEFAULT_HEADERS, ...headers },
      method,
      ...others,
    })
      .then(async (response) => {
        const json = await response.json();

        if (response.status >= 400) reject({ code: response.status, message: json.message }); // eslint-disable-line
        else resolve(json);
      }).catch(({ message = 'ERROR_CONNECTION', response } = {}) => {
        reject({ // eslint-disable-line
          code: response ? response.status : undefined,
          message,
        });
      });
  });
};
