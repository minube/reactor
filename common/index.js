import STYLE from './style';
import THEME from './theme';

export {
  STYLE,
  THEME,
};
