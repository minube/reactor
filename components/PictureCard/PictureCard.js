import { array, func, node, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { ImageBackground, Text, View } from 'react-native';

import { Touchable } from '../';
import styles from './PictureCard.style';

const IMAGE_PLACEHOLDER = 'https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg';

const PictureCard = ({
  children, caption, image, location, onPress, style, title,
}) => (
  <Touchable disabled={!onPress} onPress={onPress} style={style}>
    <ImageBackground
      resizeMode={image ? 'cover' : 'center'}
      source={{ uri: image || IMAGE_PLACEHOLDER }}
      style={[styles.container, style]}
    >
      { image !== IMAGE_PLACEHOLDER && <View pointerEvents="none" style={image && styles.opacity} /> }
      { location && <Text style={[styles.text, styles.location]}>{location.toUpperCase()}</Text> }
      <View pointerEvents="none" style={location ? styles.contentBottom : styles.content}>
        { title && <Text style={[styles.text, styles.title]}>{title}</Text> }
        { caption && <Text style={[styles.text, styles.caption]}>{caption}</Text> }
      </View>
      { children }
    </ImageBackground>
  </Touchable>
);

PictureCard.propTypes = {
  children: node,
  location: string,
  caption: string,
  image: string,
  onPress: func,
  style: oneOfType([array, number]),
  title: string,
};

PictureCard.defaultProps = {
  children: undefined,
  location: undefined,
  caption: undefined,
  image: undefined,
  onPress: undefined,
  style: [],
  title: undefined,
};

export default PictureCard;
