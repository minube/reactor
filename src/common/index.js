import C from './constants';
import STYLE from './style';
import THEME from './theme';

export {
  C,
  STYLE,
  THEME,
};
