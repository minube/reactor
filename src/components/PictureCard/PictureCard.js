import { array, bool, func, node, number, oneOfType, string } from 'prop-types';
import React from 'react';
import { ImageBackground, Text, View } from 'react-native';

import { Touchable } from '../';
import styles from './PictureCard.style';

const IMAGE_PLACEHOLDER = 'https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg';

const PictureCard = ({
  children, caption, elevation, image, location, onPress, style, title,
}) => (
  <Touchable disabled={!onPress} onPress={onPress}>
    <ImageBackground
      resizeMode={image ? 'cover' : 'center'}
      source={{ uri: image || IMAGE_PLACEHOLDER }}
      style={[styles.container, image && elevation && styles.elevation, style]}
    >
      { image !== IMAGE_PLACEHOLDER && <View pointerEvents="none" style={image && styles.opacity} /> }
      { location && <Text style={[styles.text, styles.location]}>{location.toUpperCase()}</Text> }
      <View pointerEvents="none" style={location ? styles.contentBottom : styles.content}>
        { title && <Text style={[styles.text, styles.title]}>{title}</Text> }
        { caption && <Text style={[styles.text, styles.caption]}>{caption}</Text> }
      </View>
      { children }
    </ImageBackground>
  </Touchable>
);

PictureCard.propTypes = {
  children: node,
  location: string,
  caption: string,
  elevation: bool,
  image: string,
  onPress: func,
  style: oneOfType([array, number]),
  title: string,
};

PictureCard.defaultProps = {
  children: undefined,
  location: undefined,
  caption: undefined,
  elevation: true,
  image: undefined,
  onPress: undefined,
  style: [],
  title: undefined,
};

export default PictureCard;
