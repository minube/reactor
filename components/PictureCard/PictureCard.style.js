import { StyleSheet } from 'react-native';

import { LAYOUT, THEME } from '../../common';

const { BORDER_RADIUS, COLOR, UNIT } = THEME;

export default StyleSheet.create({
  content: {
    backgroundColor: COLOR.BASE,
  },

  borderRadius: {
    borderRadius: BORDER_RADIUS,
  },

  image: {
    opacity: 0.8,
  },

  info: {
    ...LAYOUT.STYLE.CENTERED,
    position: 'absolute',
    left: 0,
    height: '100%',
    width: '100%',
    padding: UNIT,
    alignSelf: 'center',
  },

  small: {
    height: UNIT * 10,
  },

  text: {
    color: COLOR.WHITE,
    textAlign: 'center',
  },

  title: {
    marginBottom: UNIT / 2,
  },
});
