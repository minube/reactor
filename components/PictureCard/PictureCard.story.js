import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { boolean, color, object, text } from '@storybook/addon-knobs/react';

import { THEME } from '../../common';
import PictureCard from './index';

const LIPSUM_IMAGE = 'https://picsum.photos/320/200/?random';
const STYLE = { backgroundColor: 'rgba(0,255,0,1)', opacity: 0.75 };
const STYLE_CONTAINER = { backgroundColor: 'rgba(0,255,0,0.25)', padding: 10 };

storiesOf('✅ PictureCard', module)
  .add('default', () => (
    <PictureCard />
  ))
  .add('image', () => (
    <PictureCard image={LIPSUM_IMAGE} />
  ))
  .add('backgroundColor', () => (
    <PictureCard image={LIPSUM_IMAGE} backgroundColor={THEME.COLOR.PRIMARY} />
  ))
  .add('caption', () => (
    <PictureCard caption="{caption}" image={LIPSUM_IMAGE} />
  ))
  .add('title', () => (
    <PictureCard image={LIPSUM_IMAGE} title="{title}" />
  ))
  .add('caption & title', () => (
    <PictureCard caption="{caption}" image={LIPSUM_IMAGE} title="{title}" />
  ))
  .add('small', () => (
    <PictureCard caption="{caption}" small image={LIPSUM_IMAGE} title="{title}" />
  ))
  .add('portrait', () => (
    <PictureCard caption="{caption}" portrait image={LIPSUM_IMAGE} title="{title}" />
  ))
  .add('⚡ onPress', () => (
    <PictureCard image={LIPSUM_IMAGE} onPress={action('PictureCard.onPress()')} />
  ))
  .add('⚡ onLoad', () => (
    <PictureCard image={LIPSUM_IMAGE} onLoad={action('PictureCard.onLoad()')} />
  ))
  .add('style', () => (
    <PictureCard image={LIPSUM_IMAGE} style={STYLE} />
  ))
  .add('styleContainer', () => (
    <PictureCard image={LIPSUM_IMAGE} styleContainer={STYLE_CONTAINER} />
  ))
  .add('🏀 Playground', () => (
    <PictureCard
      caption={text('caption', 'minube stories')}
      backgroundColor={color('backgroundColor', undefined)}
      small={boolean('small', false)}
      image={text('image', LIPSUM_IMAGE)}
      onPress={action('PictureCard.onPress()')}
      portrait={boolean('portrait', false)}
      title={text('title', 'Una semana en Mallorca')}
      style={object('style', STYLE)}
      styleContainer={object('styleContainer', STYLE_CONTAINER)}
    />
  ));
