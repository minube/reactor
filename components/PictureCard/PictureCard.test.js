import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { THEME } from '../../common';
import PictureCard from './PictureCard';

const LIPSUM_IMAGE = 'https://picsum.photos/320/200/?random';

describe('<PictureCard>', () => {
  it('renders', () => {
    const tree = renderer.create(<PictureCard />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {image}', () => {
    const tree = renderer.create(<PictureCard image={LIPSUM_IMAGE} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {backgroundColor}', () => {
    const tree = renderer.create(<PictureCard backgroundColor={THEME.COLOR.PRIMARY} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {caption}', () => {
    const tree = renderer.create(<PictureCard caption="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {title}', () => {
    const tree = renderer.create(<PictureCard title="Hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {small}', () => {
    const tree = renderer.create(<PictureCard small />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {portrait}', () => {
    const tree = renderer.create(<PictureCard portrait />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onPress}', () => {
    // @TODO: Test <PictureCard /> {onPress}
  });

  it('when {onLoad}', () => {
    // @TODO: Test <PictureCard /> {onLoad}
  });

  it('when {style}', () => {
    const tree = renderer.create(<PictureCard style={{ backgroundColor: 'orange', opacity: 0.75 }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders in DOM environment', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PictureCard />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

