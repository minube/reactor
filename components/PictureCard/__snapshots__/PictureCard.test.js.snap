// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PictureCard> renders 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {backgroundColor} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        Object {
          "backgroundColor": "#10c0bc",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {caption} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontFamily": undefined,
              },
              false,
              false,
              false,
              false,
              Object {
                "fontSize": 16,
                "lineHeight": 20.8,
              },
              Object {
                "fontSize": 12,
                "lineHeight": 14.399999999999999,
              },
              false,
              false,
              false,
              false,
              false,
              false,
              false,
              Object {
                "color": "#ffffff",
                "textAlign": "center",
              },
            ]
          }
        >
          Hello world
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {image} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://picsum.photos/320/200/?random",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {portrait} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 220,
            },
            false,
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 220,
            },
            false,
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {small} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            Object {
              "height": 100,
            },
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            Object {
              "height": 100,
            },
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {style} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            Object {
              "backgroundColor": "orange",
              "opacity": 0.75,
            },
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            Object {
              "backgroundColor": "orange",
              "opacity": 0.75,
            },
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`<PictureCard> when {title} 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  pointerEvents="box-only"
  style={undefined}
  testID={undefined}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#CDD2D4",
        },
        Object {
          "borderRadius": 3,
        },
        Object {
          "width": 168,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#f4f4f4",
            "justifyContent": "center",
          },
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ],
        ]
      }
    >
      <Image
        onLoad={[Function]}
        resizeMode="center"
        source={
          Object {
            "uri": "https://cdn.mnstatic.com/1/svg/placeholder/eiffel_tower.svg",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 3,
            },
            Object {
              "opacity": 0.8,
            },
            Object {
              "width": 168,
            },
            Object {
              "height": 190,
            },
            false,
            undefined,
          ]
        }
      />
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
        style={
          Array [
            Object {
              "position": "absolute",
            },
            false,
          ]
        }
      />
    </View>
    <View
      pointerEvents="none"
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "alignSelf": "center",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "padding": 10,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={undefined}
          style={
            Array [
              Object {
                "color": "#4a4a4a",
                "fontFamily": undefined,
              },
              false,
              false,
              Object {
                "fontWeight": "bold",
              },
              false,
              Object {
                "fontSize": 16,
                "lineHeight": 20.8,
              },
              false,
              false,
              Object {
                "fontSize": 20,
                "lineHeight": 26,
              },
              false,
              false,
              false,
              false,
              false,
              Object {
                "color": "#ffffff",
                "marginBottom": 5,
                "textAlign": "center",
              },
            ]
          }
        >
          Hello world
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 3,
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
